### PR TITLE
[WIP]{KeyVault} KeyVault Track 2 SDK adoption

### PR DESCRIPTION
**Pull request by bim-msft**
This is for testing, **do not merge**

Python SDK PR: https://github.com/Azure/azure-sdk-for-python/pull/10128
Swagger readme: https://github.com/bim-msft/azure-rest-api-specs/blob/bim_kv_track2/specification/keyvault/resource-manager/readme.python.md

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
KeyVault Track 2

File: src/azure-cli-core/azure/cli/core/commands/client_factory.py [hunk 1]
```diff
@@ -144,7 +144,8 @@ def _get_mgmt_service_client(cli_ctx,
     else:
         client = client_type(cred, **client_kwargs)
 
-    configure_common_settings(cli_ctx, client)
+    if hasattr(client, 'config'):
+        configure_common_settings(cli_ctx, client)
```
Comment (lmazuel) [inline src/azure-cli-core/azure/cli/core/commands/client_factory.py]: You need to discuss with Yu about this

Comment (bim-msft) [reply src/azure-cli-core/azure/cli/core/commands/client_factory.py]: > You need to discuss with Yu about this

Thanks, I'll talk to him.

**Comment by bim-msft** [comment]
Closing this one.

**Pull request closed without merging**